src/basic/cgroup-util.c

```diff
@@ -1467,17 +1440,10 @@ int cg_path_get_session(const char *path, char **ret_session) {
         if (!session_id_valid(start))
                 return -ENXIO;
 
-        if (ret_session) {
-                char *rr;
-
-                rr = strdup(start);
-                if (!rr)
-                        return -ENOMEM;
-
-                *ret_session = rr;
-        }
+        if (!ret_session)
```

Comment:
This is not necessary?

Reply:
`strdup_to` requires the output param to be non-empty and we would trigger the assert without this check. (An different API which accepts NULL would be possible, but allowing a NULL output is only useful very rarely, so I think it's better to have the assert.)